argilla-server/src/argilla_server/search_engine/commons.py

```diff
@@ -647,7 +647,18 @@ def _build_text_query(dataset: Dataset, text: Optional[Union[TextQuery, str]] =
         if isinstance(text, str):
             text = TextQuery(q=text)
 
-        return es_simple_query_string(es_field_for_record_field(text.field), query=text.q)
+        if text.field:
+            field = dataset.field_by_name(text.field)
+            if field is None:
+                raise Exception(f"Field {text.field} not found in dataset {dataset.id}")
+            if field.is_chat:
+                field_name = f"{text.field}.*"
+            else:
+                field_name = text.field
```

Comment:
I was thinking you missed an `else`. Let's add a empty line so we it's easier to identify the first conditional as a guard function:

```suggestion
            field = dataset.field_by_name(text.field)
            if field is None:
                raise Exception(f"Field {text.field} not found in dataset {dataset.id}")

            if field.is_chat:
                field_name = f"{text.field}.*"
            else:
                field_name = text.field
```

Reply:
Time to see the doctor